src/ray/core_worker/core_worker.cc

```diff
@@ -2048,6 +2048,35 @@ Status CoreWorker::CancelTask(const ObjectID &object_id,
   return Status::OK();
 }
 
+Status CoreWorker::CancelMultipleTasks(const std::vector<ObjectID> &object_ids,
+                                       bool force_kill,
+                                       bool recursive) {
+  for (const ObjectID &object_id : object_ids) {
+    auto task_spec = task_manager_->GetTaskSpec(object_id.TaskId());
+    direct_task_submitter_->ClearStatusByTaskID(task_spec.value().TaskId());
+  }
+
+  for (const ObjectID &object_id : object_ids) {
+    CancelTask(object_id, force_kill, recursive);
+  }
+
+  for (const ObjectID &object_id : object_ids) {
+    auto task_spec = task_manager_->GetTaskSpec(object_id.TaskId());
+    Status temp_result =
+        direct_task_submitter_->GetStatusByTaskID(task_spec.value().TaskId());
+    int64_t wait_time = 1;
+    while (temp_result.IsNotSetYet()) {
+      std::this_thread::sleep_for(std::chrono::seconds(wait_time));
```

Comment:
this waiting might not be ideal. that makes me wonder what's the right semantics of this call. FWIW today's cancel task is already best efforts. Maybe a good enough semantics is ensure the cancel requests is delivered?  what do you think ? @jjyao @rkooo567

Reply:
> this waiting might not be ideal.

I totally agree with you. This is just something naive I implemented for getting things done. Anyways please see https://github.com/ray-project/ray/pull/24685#discussion_r874414383

Reply:
Yeah non-blocking should be okay. That's the same semantic as ray.kill or remove_placement_group. We just need to make sure the request is delivered.